engine/is_raspberrypi.py

```diff
@@ -0,0 +1,13 @@
+import io
+import os
+"""a function from : https://raspberrypi.stackexchange.com/questions/5100/detect-that-a-python-program-is-running-on-the-pi/74541#74541
+return True if the current device is a raspberry_pi, False otherwise"""
+def is_raspberrypi():
+    try:
```

Comment:
```suggestion
def is_raspberrypi():
    """
    Return True if the current device is a raspberry_pi, False otherwise.
    Reference: https://raspberrypi.stackexchange.com/questions/5100/detect-that-a-python-program-is-running-on-the-pi/74541#74541
    """
    try:
```

Reply:
oops i dont think this suggestion will work bc its not recognizing the multiline comment, but think we can move the spec under the function header + I changed a function from to Reference